## PCR

### Summary
This is a procedure for polymerase chain reaction (PCR). This protocol was designed specifically for Sanger sequencing. Each study will require fine tuning of this protocol as PCR protocols may not transfer well from one study to another. This protocol provides a starting point but may need modifications (see PCR optimization procedure and troubleshooting).

### Equipment & Supplies
- Vortex
- Pipettes
- Pipette tips
- Microcentrifuge tubes
- PCR tubes
- Table-top spinner
- Thermal cycler

### Reagents & Standards
- Template DNA (~10 ng/µL)
- Taq Master Mix (GoTaq Master Mix, Promega)
- Forward & reverse primers
- Molecular-grade H2O

### Sample Preparation & Storage
All reagents should be kept frozen until you are ready to use. Repeated freezing and thawing will kill Taq and primers. The other materials may be thawed and allowed to set out during preparation.

### Procedure
To minimize pipette error and chances of contamination, you will first make a master mix solution depending on how many samples you are running plus 2 (one for a negative control and one to account for pipetting error). This master mix solution is then put into each tube so that you are not putting in each reagent into each tube separately.

#### PCR for Sanger Sequencing
1.	Thaw all reagents and keep them on ice. Briefly vortex and spin to collect the material at the bottom of the tubes.
2.	Make one of the following PCR reaction mix solutions on ice.

The followings are an example for 16 samples in 20-µL and 10-µL reaction volumes.
For 20-µL reaction volume:
|Reagents	|Amount| Final concentration| Number of samples | Total volume|
|:--------|:-----|:-------------------|:------------------|:------------|
|GoTaq Master Mix	|10 µL| 1X| | 200 µL|
|Forward primer (10 µM)	|1 µL| 0.1 – 1.0 µM| | 20 µL|
|Reverse primer	(10 µM) |1 µL| 0.1 – 1.0 µM| x 20| 20 µL|
|Molecular grade H2O	|6 µL| N.A.| |  120 µL|
|DNA template (10 ng/µL) |2 µL| <250 ng| | N.A|

For 10-µL reaction volume:
|Reagents	|Amount| Final concentration|
|:--------|:-----|:-------------------|
|Taq Master Mix	|5 µL| 1X|
|Forward primer (10 µM)	|0.5 µL| 0.1 – 1.0 µM|
|Reverse primer	(10 µM) |0.5 µL| 0.1 – 1.0 µM|
|Molecular grade H2O	|3 µL| N.A.|
|DNA template (10 ng/µL) |1 µL| <250 ng|


Make enough for # of samples plus 2 extra (1 for negative control, 1 for pipetting error). 
For example, you will make PCR mix solution for 10 samples if you have 8 samples (plus 1 for negative control, 1 for pipetting error).

2.	Add 18µL of master mix to each labeled PCR tube.
3.	Aliquot 2µL DNA into properly labeled, small PCR tubes. Use a new tip for each sample. Run one sample with DNA but only master mix. Label this tube as X or NC for negative control. (If you get amplification in the negative control, something has been contaminated).
4.	Remove any bubbles in the tubes by spinning in the fish-eye centrifuge. If this does not work, try popping bubbles with a small pipette tip.
5.	Close tubes and place them into the thermal cycler and run program. (Program will vary with project and gene being amplified).


### CLEAN UP

### DATA ANALYSIS AND CALCULATIONS
PCR products can be checked by agarose gel electrophoresis.

### NOTES
See [PCR cleanup (EXOSAP)](PCR_exosap.md) and [Cycle Sequencing](Cycle_sequencing.md) for next steps in Sanger sequencing process.
