#PCR
##SUMMARY
This is a procedure for amplification of gene segments. This protocol was designed specifically for mtDNA gene and microsatellite locus amplification. Each study will require fine tuning of this protocol as PCR protocols may not transfer well from one study to another. This protocol provides a starting point but may need modifications (see PCR optimization procedure and troubleshooting).

EQUIPMENT AND SUPPLIES
Thermal cycler, gel electrophoresis materials, fish-eye centrifuge, PCR tubes

REAGENTS AND STANDARDS
Template DNA (~10ng/µL)
Taq Master Mix (Promega, Qiagen, etc.)
Forward primer
Reverse primer
Molecular grade H2O
MgCl2 (optional)

SAMPLE PREPARATION AND STORAGE
All reagents should be kept frozen until you are ready to use. Taq Master Mix should remain in the freezer until you are ready to add it, as excessive freezing and thawing will kill Taq. The other materials may be thawed and allowed to set out during preparation.

PROCEDURE
To minimize pipette error and chances of contamination, you will first make a master mix solution depending on how many specimens you are running plus 2 (one for a negative control and one to account for pipetting error). This master mix solution is then put into each tube so that you are not putting in each reagent into each tube separately.

PCR for Sequencing
1.	Make PCR master mix solution
PCR reagents are Progmega or Qiagen brand and are stored in an orange box in the freezer. Thaw reagents in your hand by shaking. DO NOT heat. Keeping them on ice is highly recommended

For 20µL reaction, in each PCR tube:
Reagents	Amount
Taq Master Mix	10 µL
Forward primer	1 µL
Reverse primer	1 µL
Molecular grade H2O	6 µL
DNA	2 µL
Total	20 µL
Make enough for # of samples plus 2 extra (1 for negative control, 1 for pipetting error). 
For example, you will make PCR mix solution for 10 samples if you have 8 samples (plus 1 for negative control, 1 for pipetting error).

2.	Add 18µL of master mix to each labeled PCR tube.
3.	Aliquot 2µL DNA into properly labeled, small PCR tubes. Use a new tip for each sample. Run one sample with DNA but only master mix. Label this tube as X or NC for negative control. (If you get amplification in the negative control, something has been contaminated).
4.	Remove any bubbles in the tubes by spinning in the fish-eye centrifuge. If this does not work, try popping bubbles with a small pipette tip.
5.	Close tubes and place them into the thermal cycler and run program. (Program will vary with project and gene being amplified).

PCR for Genotyping
1.	Set up a PCR as above, except one of microsatellite primers is labeled with fluorescent tag (e.g., 6-FAM, VIC, PET, NED). Since fluorescent tags are sensitive to light, DO NOT expose primers under light for long time. Keeping them on ice is highly recommended. 
2.	Follow step 1 through 5 on PCR for Sequencing.
3.	Thermal cycles may be differed in each primer set. Check appropriate thermal cycle for each primer set. 

Basic thermal cycle:
Initial denaturing at 95˚C for 2 min, followed by 35 cycles with denature at 95˚C for 30 s, annealing at 52-55˚C for 30 s, extension at 72˚C for 1 min, and final extension at 72˚C for 10 min.

CLEAN UP

DATA ANALYSIS AND CALCULATIONS
PCR products can be checked by agarose gel electrophoresis.

NOTES
